src/SablierV2MerkleLockupFactory.sol

```diff
@@ -51,4 +56,53 @@ contract SablierV2MerkleLockupFactory is ISablierV2MerkleLockupFactory {
             merkleLockupLL, baseParams, lockupLinear, streamDurations, aggregateAmount, recipientsCount
         );
     }
+
+    /// @notice inheritdoc ISablierV2MerkleLockupFactory
+    function createMerkleLockupLT(
+        MerkleLockup.ConstructorParams memory baseParams,
+        ISablierV2LockupTranched lockupTranched,
+        MerkleLockupLT.TrancheWithPercentage[] memory tranchesWithPercentages,
+        uint256 aggregateAmount,
+        uint256 recipientsCount
+    )
+        external
+        returns (ISablierV2MerkleLockupLT merkleLockupLT)
+    {
+        // Calculate the sum of percentages across all tranches.
+        UD60x18 percentagesSum;
+        uint256 trancheCount = tranchesWithPercentages.length;
+        for (uint256 i = 0; i < trancheCount; ++i) {
+            UD60x18 percentage = (tranchesWithPercentages[i].amountPercentage).intoUD60x18();
+            percentagesSum = percentagesSum.add(percentage);
+        }
+
+        // Checks: the sum percentage equal 100%.
+        if (!percentagesSum.eq(ud(1e18))) {
+            revert Errors.SablierV2MerkleLockupFactory_PercentageSumNotEqualOneHundred(percentagesSum.intoUint256());
+        }
+
+        // Hash the parameters to generate a salt.
+        bytes32 salt = keccak256(
+            abi.encodePacked(
+                baseParams.initialAdmin,
+                baseParams.asset,
+                abi.encode(baseParams.ipfsCID),
+                bytes32(abi.encodePacked(baseParams.name)),
+                baseParams.merkleRoot,
+                baseParams.expiration,
+                baseParams.cancelable,
+                baseParams.transferable,
+                lockupTranched,
+                abi.encode(tranchesWithPercentages)
+            )
+        );
+
+        // Deploy the Merkle Lockup contract with CREATE2.
+        merkleLockupLT = new SablierV2MerkleLockupLT{ salt: salt }(baseParams, lockupTranched, tranchesWithPercentages);
+
+        // Log the creation of the Merkle Lockup, including some metadata that is not stored on-chain.
```

Comment:
We can remove "including some metadata that is not stored on-chain" now since all the data is being stored on chain.

Reply:
Done in https://github.com/sablier-labs/v2-periphery/pull/301/commits/c1c7b0bb699c2bde672c86fb4b4cdb498db84e09

Reply:
not all data is stored on chain:

`aggregateAmount` and `recipientsCount` isn't

Reply:
Yes you are correct. I was focussing on ipfsCID only.